client/src/includes/tabs.js

```diff
@@ -0,0 +1,371 @@
+
+    // Tab Options - Add these data attributes along side the data-tabs attribute
+    // Use this to enable fade-in animations on tab select
+    this.animate = this.tabContainer.hasAttribute('data-tabs-animate');
+    // Disable url hash from appearing on tab select (normally used in modals)
+    this.disableUrl = this.tabContainer.hasAttribute('data-tabs-disable-url');
+
+    this.state = {
+      // Tab Settings
+      activeTabID: '',
+      transition: 150,
+      // CSS Classes
+      css: {
+        animate: 'animate-in',
+      },
+      // Keyboard Keys
+      keys: {
+        end: 'End',
+        home: 'Home',
+        left: 'ArrowLeft',
+        up: 'ArrowUp',
+        right: 'ArrowRight',
+        down: 'ArrowDown',
+        enter: 'Enter',
+        space: ' ',
+        // Key names specific for Edge
+        edgeBrowser: {
+          left: 'Left',
+          right: 'Right',
+          down: 'Down',
+          up: 'Up',
+        },
+      },
+      direction: {
+        ArrowLeft: -1,
+        Left: -1,
+        ArrowUp: -1,
+        Up: -1,
+        ArrowRight: 1,
+        Right: 1,
+        ArrowDown: 1,
+        Down: 1,
+      },
+    };
+
+    this.onComponentLoaded();
+  }
+
+  onComponentLoaded() {
+    this.bindEvents();
+
+    // Set active tab from url or make first tab active
+    if (this.tabButtons) {
+      // Set each button's aria-controls attribute and select tab if aria-selected has already been set on the element
+      this.tabButtons.forEach((button) => {
+        button.setAttribute(
+          'aria-controls',
+          button.getAttribute('href').substring(1),
+        );
+      });
+
+      // Check for active items set by the template
+      const tabActive = [...this.tabButtons].find(
+        (button) => button.getAttribute('aria-selected') === 'true',
+      );
+
+      if (window.location.hash && !this.disableUrl) {
+        this.selectTabByURLHash();
+      } else if (tabActive) {
+        // If a tab isn't hidden for some reason hide it
+        this.tabPanels.forEach((tab) => {
+          // eslint-disable-next-line no-param-reassign
+          tab.hidden = true;
+        });
+        // Show aria-selected tab
+        this.selectTab(tabActive);
+      } else {
+        this.selectFirstTab();
+      }
+    }
+  }
+
+  /**
+   * @param {string}newTabId
+   */
+  unSelectActiveTab(newTabId) {
+    // IF new tab ID is the current then don't transition out
+    if (newTabId === this.state.activeTabID || !this.state.activeTabID) {
+      return;
+    }
+
+    // Tab Content to deactivate
+    const tabContent = this.tabContainer.querySelector(
+      `#${this.state.activeTabID}`,
+    );
+
+    if (this.animate) {
+      this.animateOut(tabContent);
+    } else {
+      tabContent.hidden = true;
+    }
+
+    const tab = this.tabContainer.querySelector(
+      `a[href='#${this.state.activeTabID}']`,
+    );
+
+    tab.setAttribute('aria-selected', 'false');
+    tab.setAttribute('tabindex', '-1');
```

Comment:
Setting this on select and removing on de-select allows us to arrive back at the currently active tab when we tab back to the nav after tabbing away. This is also used to prevent us to "tab key" to the next tab in the list so it helps for both scenarios i believe 👀